src/Plugin/EntityReferenceSelection/OgSelection.php

```diff
@@ -77,37 +139,45 @@ protected function buildEntityQuery($match = NULL, $match_operator = 'CONTAINS')
       $query->condition($bundle_key, $bundles, 'IN');
     }
 
-    $user_groups = $this->getUserGroups();
-    if (!$user_groups) {
+    // Get the identifier key of the entity.
+    $identifier_key = $definition->getKey('id');
+
+    if ($this->currentUser->isAnonymous()) {
+      // @todo: Check if anonymous users should have access to any referenced
+      // groups? What about groups that allow anonymous posts?
+      return $query->condition($identifier_key, -1);
+    }
+
+    if ($this->currentUser->hasPermission('administer group')) {
+      // User can see all the groups.
       return $query;
     }
 
-    $identifier_key = $definition->getKey('id');
+    if (empty($this->configuration['entity'])) {
+      // @todo: Find out why we have this scenario.
+      return $query;
+    }
+
+    /** @var \Drupal\Core\Entity\EntityInterface $entity */
+    $entity = $this->configuration['entity'];
+    $entity_type_id = $entity->getEntityTypeId();
+    $bundle = $entity->bundle();
 
     $ids = [];
-    if (!empty($this->configuration['handler_settings']['field_mode']) && $this->configuration['handler_settings']['field_mode'] == 'admin') {
-      // Don't include the groups, the user doesn't have create permission.
-      foreach ($user_groups as $delta => $group) {
+    foreach ($this->getUserGroups() as $group) {
```

Comment:
Currently it is possible to allow non-members to create group content. This makes it impossible for them to do so.

While an edge scenario, we have a requirement to allow organisation members to post specific type of content to any interest group, even ones they are not a member of.

There are other ways we can work around the limitation (e.g. alter the group content form to inject a multi-select field that list all public groups, then programatically update the `og_audience` field in pre-submit. But this feels rather hacky.

Any thoughts?

Reply:
> Currently it is possible to allow non-members to create group content

Yes, that use case should remain.